src/lib.rs

```diff
@@ -173,6 +240,21 @@ pub trait Integer: Sized + Num + PartialOrd + Ord + Eq {
     }
 }
 
+/// Greatest common divisor and Bézout coefficients
+#[derive(Debug, Copy, PartialEq, Eq)]
+#[cfg_attr(array_clone, derive(Clone))]
+pub struct ExtendedGcd<A> {
+    pub gcd: A,
+    pub coeffs: [A; 2],
```

Comment:
Let's use separate `x` and `y`, so we can document it something like [this](https://en.wikipedia.org/wiki/B%C3%A9zout%27s_identity):

> Bézout's identity — Let a and b be integers with greatest common divisor d. Then, there exist integers x and y such that ax + by = d.

Then we also don't have to worry about `array_clone`.